src/Polysemy/Scoped.hs

```diff
@@ -36,26 +36,26 @@ import Polysemy.Internal.Tactics
 -- use the 'Tactical' environment and transforms the effect into other effects
 -- on the stack.
 interpretScopedH ::
-  ∀ param resource effect r .
+  ∀ resource param effect r .
   -- | A callback function that allows the user to acquire a resource for each
   -- computation wrapped by 'scoped' using other effects, with an additional
   -- argument that contains the call site parameter passed to 'scoped'.
   (∀ x . param -> (resource -> Sem r x) -> Sem r x) ->
   -- | A handler like the one expected by 'interpretH' with an additional
   -- parameter that contains the @resource@ allocated by the first argument.
   (∀ r0 x . resource -> effect (Sem r0) x -> Tactical effect (Sem r0) r x) ->
-  InterpreterFor (Scoped param resource effect) r
+  InterpreterFor (Scoped param effect) r
 interpretScopedH withResource scopedHandler =
-  go
+  go (errorWithoutStackTrace "top level run")
```

Comment:
Leave a TODO here for us to investigate whether the loopbreaker optimization is worth it here (i.e. this pattern):
```
go :: ...
go = ... // same as existing but replace all recursive references to go with go1
{-# INLINE go #-}
go1 :: ...
go1 = go
{-# NOINLINE go #-}
```
Alternatively, if you know it's not worth it, say so and don't add that TODO.